apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/Adapters.kt

```diff
@@ -81,9 +82,31 @@ class OptionalAdapter<T>(private val wrappedAdapter: Adapter<T>) : Adapter<Optio
   }
 }
 
-@SharedImmutable
+/**
+ * This adapter is used to handle nullable fields when they are represented as [Optional].
+ * `null` is deserialized as [Optional.Absent].
+ */
+class OptionalFieldAdapter<T>(private val wrappedAdapter: Adapter<T>) : Adapter<Optional<@JvmSuppressWildcards T>> {
```

Comment:
Not 100% happy about the naming `OptionalFieldAdapter` but can't think of something better to differentiate with existing `OptionalAdapter`.

Reply:
I'd vote for `ApolloOptionalAdapter` (and `PresentAdapter` for the other one in v4)